File: fsharp/high-scores/HighScores.fs
module HighScores

// Assume list of scores is ordered by time of entry.

/// List all of the scores
let scores (values: int list): int list = values

/// Return the last score
let latest (values: int list): int = List.last values

/// Return the top score
let personalBest (values: int list): int = List.max values

/// Return the top three scores
let personalTopThree (values: int list): int list =
    let values = List.sortDescending values
    if values.Length < 3 then values
    else List.take 3 values

File: java/high-scores/src/main/java/HighScores.java
import java.util.ArrayList;
import java.util.List;
import java.util.Collections;

class HighScores {
    private final List<Integer> highScores;

    public HighScores(List<Integer> highScores) {
        this.highScores = highScores;
    }

    List<Integer> scores() {
        return List.copyOf(highScores);
    }

    Integer latest() {
        return highScores.getLast();
    }

    Integer personalBest() {
        return Collections.max(highScores);
    }

    List<Integer> personalTopThree() {
        List<Integer> scores = new ArrayList<>();
        scores.addAll(highScores);
        scores.sort(Collections.reverseOrder());
        int index = Math.min(3, scores.size());
        return scores.subList(0, index);
    }

}
